pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py

```diff
@@ -1,202 +1,223 @@
+            Initialisation procedure of the detector.
+
+            Returns
+            -------
+
+                The initialized status.
+
+            See Also
+            --------
+            daq_utils.ThreadCommand
+        """
+        self.status.update(edict(initialized = False, info = "", x_axis = None, y_axis = None, controller = None))
+        try:
+            if self.settings.child(('controller_status')).value() == "Slave":
+                if controller is None:
+                    raise Exception('no controller has been defined externally while this detector is a slave one')
+                else:
+                    self.controller = controller
+            else:
+                self.controller = active_dso
+                usb_address = "USBTMC:" + self.settings.child(('VISA_ressources')).value()
+                self.controller.MakeConnection(usb_address)
+                # set the timeout of the scope to 10 seconds
+                # may be not needed
+                self.controller.setTimeout(10)
+
+            self.status.initialized = True
+            self.status.controller = self.controller
+            # self.controller.WriteString(r"""vbs 'app.acquisition.triggermode = "stopped" ' """, 1)
+            return self.status
+        except Exception as e:
+            self.emit_status(ThreadCommand('Update_Status', [getLineInfo() + str(e), 'log']))
+            self.status.info = getLineInfo() + str(e)
+            self.status.initialized = False
+            return self.status
+
+    def commit_settings(self, param):
+        """
+            Activate the parameters changes in the hardware.
+
+            =============== ================================= ============================
+            **Parameters**   **Type**                         **Description**
+            *param*         instance of pyqtgraph.parameter   The parameter to be checked.
+            =============== ================================= ============================
+
+            See Also
+            --------
+            daq_utils.ThreadCommand
+        """
+        try:
+            pass
+        except Exception as e:
+            self.emit_status(ThreadCommand('Update_Status',[getLineInfo() + str(e),'log']))
+
+    def grab_data(self, Naverage=1, **kwargs):
+        """
+            | Start new acquisition.
+            | Grab the current values.
+            | Send the data_grabed_signal once done.
+
+            =============== ======== ===============================================
+            **Parameters**  **Type**  **Description**
+            *Naverage*      int       Number of values to average
+            =============== ======== ===============================================
+        """
+        channel = self.settings.child(('channels')).value()['selected']
+
+        self.controller.WriteString(r"""vbs? 'return=app.acquisition.horizontal.numsegments' """, 1)
+        numSegments = self.controller.ReadString(8)

```

Comment:
The numSegments variable (maybe "numberOfSegments" would be more explicit…) seems to be used only in the if sampleMode == "Sequence" condition so I would place those lines in it.